modules/data-store/test/data-store.test.js

```diff
@@ -0,0 +1,239 @@
+        expect(dataStore[ENTITIES_BY_UUID][entity.id]).toBe(dataStore[ENTITIES_BY_PATH]["/tata"][index]);
+        expect(dataStore[ENTITIES_BY_PATH]["/tata"][index]).toBe(entity);
+    });
+
+    test("entity path is correct", () => {
+        const dataStore = new DataStore();
+        const entity = new Entity({ id: "entity-0" });
+        dataStore.addEntity(entity, "/tata");
+        expect(entity.getPath()).toBe("/tata");
+    });
+
+});
+
+describe("DataStore.removeEntity", () => {
+
+    test("entity is removed from store", () => {
+        const dataStore = new DataStore();
+        const entity = new Entity({ id: "entity-0" });
+        const entity1 = new Entity({ id: "entity-1" });
+        dataStore.addEntity(entity, "/tata");
+        dataStore.addEntity(entity1, "/tata");
+        dataStore.removeEntity(entity.id);
+        expect(dataStore[ENTITIES_BY_PATH]["/tata"].indexOf(entity)).toEqual(-1);
+    });
+
+    test("path is removed if empty", () => {
+        const dataStore = new DataStore();
+        const entity = new Entity({ id: "entity-0" });
+        const entity1 = new Entity({ id: "entity-1" });
+        dataStore.addEntity(entity, "/tata");
+        dataStore.addEntity(entity1, "/tata");
+        dataStore.removeEntity(entity.id);
+        dataStore.removeEntity(entity1.id);
+        expect(dataStore[ENTITIES_BY_PATH]["/tata"]).toBeUndefined();
+    });
+
+    test("uuid is removed", () => {
+        const dataStore = new DataStore();
+        const entity = new Entity({ id: "entity-0" });
+        dataStore.addEntity(entity, "/tata");
+        dataStore.removeEntity(entity.id);
+        expect(dataStore[ENTITIES_BY_UUID][entity.id]).toBeUndefined();
+    });
+
+    test("entity path is removed", () => {
+        const dataStore = new DataStore();
+        const entity = new Entity({ id: "entity-0" });
+        dataStore.addEntity(entity, "/tata");
+        dataStore.removeEntity(entity.id);
+        expect(entity.getPath()).toBeUndefined();
+    });
+
+    test("can remove entity by giving entity in parameters", () => {
+        const dataStore = new DataStore();
+        const entity = new Entity({ id: "entity-0" });
+        const entity1 = new Entity({ id: "entity-1" });
+        dataStore.addEntity(entity, "/tata");
+        dataStore.addEntity(entity1, "/tata");
+        dataStore.removeEntity(entity);
+        expect(dataStore[ENTITIES_BY_PATH]["/tata"].indexOf(entity)).toEqual(-1);
+        expect(dataStore[ENTITIES_BY_UUID][entity.id]).toBeUndefined();
+        dataStore.removeEntity(entity1);
+        expect(dataStore[ENTITIES_BY_PATH]["/tata"]).toBeUndefined();
+    });
+
+});
+
+describe("DataStore.getEntity", () => {
+
+    test("entity exists", () => {
+        const dataStore = new DataStore();
+        const entity = new Entity({ id: "entity-0" });
+        dataStore.addEntity(entity);
+        expect(dataStore.getEntity(entity.id)).toBe(entity);
+    });
+
+    test("path is correct", () => {
```

Comment:
This test is not related to getEntity(), it should (and it is already) be tested in the addEntity() section